ciao-controller/instance.go

```diff
@@ -193,70 +214,27 @@ func getStorage(c *controller, wl *types.Workload, tenant string, instanceID str
 			return payloads.StorageResource{}, err
 		}
 
-		// don't you need to add support for indicating whether
-		// a block device is bootable.
-		data := types.BlockData{
-			BlockDevice: device,
-			Size:        s.Size,
-			CreateTime:  time.Now(),
-			TenantID:    tenant,
-			Name:        fmt.Sprintf("Storage for instance: %s", instanceID),
-		}
-
-		err = c.ds.AddBlockDevice(data)
-		if err != nil {
-			c.DeleteBlockDevice(device.ID)
-			return payloads.StorageResource{}, err
-		}
+		s.Persistent = true
```

Comment:
when you modify s you are actually modifying the cached workload since this is by reference, thus preserving your changes. for the next user of this workload. Is that ok?